src/main/scripts/upgrade_oracle_5_0.sql

```diff
@@ -0,0 +1,205 @@
+-- Add Technique table
+
+CREATE TABLE TECHNIQUE (
+	ID NUMBER(19) NOT NULL,
+	CREATE_ID VARCHAR2(255) NOT NULL,
+	CREATE_TIME TIMESTAMP NOT NULL,
+	DESCRIPTION VARCHAR2(255),
+	MOD_ID VARCHAR2(255) NOT NULL,
+	MOD_TIME TIMESTAMP NOT NULL,
+	NAME VARCHAR2(255) NOT NULL,
+	PID VARCHAR2(255),
+	PRIMARY KEY (ID)
+);
+
+ALTER TABLE TECHNIQUE ADD CONSTRAINT UNQ_TECHNIQUE_0 UNIQUE (NAME);
+
+-- Add DatasetTechnique table
+
+CREATE TABLE DATASETTECHNIQUE (
+	ID NUMBER(19) NOT NULL,
+	CREATE_ID VARCHAR2(255) NOT NULL,
+	CREATE_TIME TIMESTAMP NOT NULL,
+	MOD_ID VARCHAR2(255) NOT NULL,
+	MOD_TIME TIMESTAMP NOT NULL,
+	DATASET_ID NUMBER(19) NOT NULL,
+	TECHNIQUE_ID NUMBER(19) NOT NULL,
+	PRIMARY KEY (ID)
+);
+
+ALTER TABLE DATASETTECHNIQUE ADD CONSTRAINT FK_DATASETTECHNIQUE_DATASET_ID FOREIGN KEY (DATASET_ID) REFERENCES DATASET (ID);
+ALTER TABLE DATASETTECHNIQUE ADD CONSTRAINT FK_DATASETTECHNIQUE_TECHNIQUE_ID FOREIGN KEY (TECHNIQUE_ID) REFERENCES TECHNIQUE (ID);
```

Comment:
I don't think we specify anywhere what versions of Oracle ICAT supports but if we want to remain compatible with versions prior to 12.2 where the length of an Oracle identifier was increased from 30 to 128 bytes, then we will need to reduce the length of some of these constraint names. From a quick bit of googling and reading of https://en.wikipedia.org/wiki/Oracle_Database and https://support.oracle.com/knowledge/Oracle%20Database%20Products/742060_1.html it looks like 11.2 and 12.1 are still being supported.

Reply:
I don't mind changing that.  But this is only the upgrade script.  If you do a fresh install the database initialization code that JPA implicitly generates from the Java Source will use the same constraint names (I believe).

Reply:
It's only the upgrade script that needs changing. For a fresh install, Eclipselink is presumably aware of this limitation in Oracle because it reduces the length of the constraint name accordingly. 

Out of interest, it seems to drop the "FK" from the beginning, drop the underscores and then start dropping vowels until the length is 30 characters. 

I could copy the autogenerated names that Eclipselink chooses into our upgrade script, if you like, or we could try to give them better names ourselves. I don't think it matters too much either way, but let me know if you have a preference.

Reply:
Actually, just thinking about this a bit more, I think I prefer the option to use the Eclipselink generated names. That way, if we need to write upgrade scripts in the future that modify any of the constraints then at least we know what name they should have in all ICATs (freshly installed ones or upgraded ones).

Reply:
I'd suggest that we should try to keep the result of the upgrade script as close as possible to a fresh install.  Since I do not test on Oracle myself, if you could change the upgrade script for Oracle to match the autogenerated names that would be great.

Reply:
👍

Reply:
I have now made these changes on the `icat-schema-extension` branch which we are looking to merge here.